waict-signalling.md

```diff
@@ -1,55 +1,143 @@
+# WAICT Enforcement Signaling
 
 # Introduction
 
-This spec describes the ways a client signals to a server that they support WAICT, and vice-versa.
+This spec describes the ways a server signals to a client to enforce the use of WAICT.
 
-# Client-side Signalling
+# Conventions
 
-Clients MAY signal that they support transparency. Doing so will allow the server to avoid sending unnecessary transparency information to the client. To signal transparency support, the client includes the request header `WAICT-Transparency-Supported: 1` in their requests to the site. Future versions of this specification may define different version numbers.
+This document uses Structured Field Values for HTTP ([RFC 9651](https://www.rfc-editor.org/rfc/rfc9651)) such as `sf-list` and `sf-integer`.
 
-# Server-side Signalling
+In this document, `origin` refers to the tuple (scheme, host, port) as defined in [RFC 6454](https://www.rfc-editor.org/rfc/rfc6454).
 
-Sites MAY to signal to the client the parameters of its transparency guarantees.
+# Advertising WAICT Support
 
-## Time-bound Signaling
+Clients SHOULD signal that they support WAICT to the server through the use of user-agent client hints. Doing so will allow the server to avoid sending unnecessary information to clients which don't support WAICT.
 
-The server MAY signal when transparency expires and where to find the inclusion proof. This is done via a response header:
-```
-WAICT-Transparency: expires=<uint64>, inclusion=<str>
-```
-where the value of the `expires` field is Unix time in seconds, the value of `inclusion` is a base64url-encoded URL which, when GETted, returns an `application/octet-stream`-encoded `EntryAndProof`. (TODO: add an option to embed inclusion into the header if it's short enough; also proof of non-inclusion or proof of tombstone inclusion to show that the site is unenrolled) (TODO: you don't need proof of non-inclusion if you just make sure your tombstone proof validity period is longer than the validity period of whatever is making the user believe transparency should be enabled)
+To signal WAICT support, the [user agent client hint](https://wicg.github.io/ua-client-hints/) `Sec-CH-WAICT` is used whose value is a `sf-list` of `sf-integers`. Each integer represents a supported version of WAICT. This specification defines version `1`. If clients include a `Sec-CH-WAICT` header in their requests, the included version numbers MUST be supported by the client.
 
-Note: The inclusion proof depends on the manifest. To ensure that all data is coherent, the URLs SHOULD include some component that is unique to the site version, e.g., the current site history hash, or the integrity policy hash.
+Servers supporting WAICT SHOULD actively solicit client hints for WAICT by including `Sec-CH-WAICT` in their `Accept-CH` response header (See [Section 3.1 of RFC 8942](https://www.rfc-editor.org/rfc/rfc8942#section-3.1)). Servers MUST tolerate unknown integers in the `Sec-CH-WAICT` header.
```

Comment:
To disambiguate, maybe: "MUST tolerate unknown integers in the `Sec-CH-WAICT` **request** header"